charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -217,24 +228,26 @@ def __init__(self, provider_charm, relation_name: str = DEFAULT_RELATION_NAME):
         self.provider_charm = provider_charm
         self.relation_name = relation_name
 
-    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str) -> None:
-        """Updates the relation data bag with data from the localGateway.
+    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str, gateway_up: bool = True) -> None:
+        """Updates the relation data bag of any related applications with data from the localGateway.
+
+        This method will complete successfully even if there are no related applications.
```

Comment:
When we discussed about this one, I was thinking more of the docstring of the library. It is very important to let people know that `send_gateway_...()` will not say anything if there are no relations, and it's responsibility of the user of this library to add some sort of check in the requirer if they depend on this data. I'd add a paragraph and an example in the usage section.

Reply:
Added a note in the lib docstring, look good to you?